src/app/reporting/LoggingManagement.h

```diff
@@ -0,0 +1,571 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    Copyright (c) 2015-2017 Nest Labs, Inc.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file
+ *
+ * @brief
+ *   Management of the CHIP Event Logging.
+ *
+ */
+#pragma once
+
+#include "EventLoggingDelegate.h"
+#include "EventLoggingTypes.h"
+#include <core/CHIPCircularTLVBuffer.h>
+#include <messaging/ExchangeMgr.h>
+#include <support/PersistedCounter.h>
+#include <util/basic-types.h>
+
+#define CHIP_CONFIG_EVENT_SIZE_RESERVE 196
+
+namespace chip {
+namespace app {
+namespace reporting {
+
+/**
+ * @brief
+ *   Internal event buffer, built around the chip::TLV::CHIPCircularTLVBuffer
+ */
+
+class CircularEventBuffer : public TLV::CHIPCircularTLVBuffer
```

Comment:
Is this public API?

If not, given that we only store pointers to it in members of the classes that _are_ public API can we forward-declare it here and move the actual declaration into the .cpp ?

Reply:
no, just move it to cpp, thanks!